### PR TITLE
Adding search by action and exports

### LINKED ISSUE #5827: We need a way to export data from the Journal Log
In light of the recent fraud reports, we need a way export transactions by user from the journal history module.  This would help in creating fraud reports in the future.  Specifically, it would be nice to be able to download the data in CSV or Excel format.

### PR DESCRIPTION
**Pull request by mbayopanda**
This PR adds:
- Export option for CSV and Excel
- Search by type of action (delete or edit)

Close #5827

### CODE REVIEW

File: server/config/routes.js [hunk 1]
```diff
@@ -433,6 +433,7 @@ exports.configure = function configure(app) {
   app.get('/reports/finance/account_report', financeReports.reportAccounts.document);
   app.get('/reports/finance/account_report_multiple', financeReports.reportAccountsMultiple.document);
   app.get('/reports/finance/journal', financeReports.journal.postingReport);
+  app.get('/reports/finance/journal/log', financeReports.journal.journalLogReport);
```
Comment (jniles) [inline server/config/routes.js:436]: :+1:

**Review by jniles — COMMENTED** [review]
The CSV export works well 👍 

Could you do some updates to the XLSX export to make it more useful?  Users will want access to the information in the `value` section, but it currently isn't usable to them.

**Review by jniles — CHANGES_REQUESTED** [review]
Have you tested this with the vanga database?  It crashes as is, but when I fixed the crash, it doesn't properly display all the information:

image

Since they are the ones who will use this information, let's try to fix this before merge.

**Comment by mbayopanda** [comment]
> 
> 
> Have you tested this with the vanga database? It crashes as is, but when I fixed the crash, it doesn't properly display all the information:
> 
> image
> 
> Since they are the ones who will use this information, let's try to fix this before merge.

Thank you @jniles for the review and the catch

**Comment by mbayopanda** [comment]
@jniles changes landed

**Comment by jniles** [comment]
Hmm... still not getting any descriptions from the Vanga database in the Excel export.  Let me see if I can work on it.

**Comment by jniles** [comment]
Okay, got it working:

image

image


@mbayopanda take a look at my changes.  If they are good with you, we can merge this.

**Comment by mbayopanda** [comment]
It's LGTM 👍🏽

**Review by jniles — APPROVED** [review]
LGTM.

Bors r+

**Pull request merged**